src/backends/libtorch_c/torch_extensions/torch_redis.cpp

```diff
@@ -16,54 +18,125 @@ torch::IValue IValueFromRedisReply(RedisModuleCallReply *reply){
             return torch::IValue(str.substr(0,len));
         }
         case REDISMODULE_REPLY_INTEGER: {
-            int intValue = RedisModule_CallReplyInteger(reply);
+            int intValue = (int)RedisModule_CallReplyInteger(reply);
             return torch::IValue(intValue);
         }
         case REDISMODULE_REPLY_ARRAY: {
             c10::impl::GenericList vec = c10::impl::GenericList(c10::AnyType::create());
             size_t len = RedisModule_CallReplyLength(reply);
             for (auto i = 0; i < len; ++i) {
                 RedisModuleCallReply *subReply = RedisModule_CallReplyArrayElement(reply, i);
-                torch::IValue value = IValueFromRedisReply(subReply);
+                torch::IValue value = IValueFromRedisReply(ctx, subReply);
                 vec.push_back(value);
             }
             return torch::IValue(vec);
         }
         case REDISMODULE_REPLY_ERROR: {
             size_t len;
             const char *replyStr = RedisModule_CallReplyStringPtr(reply, &len);
-            throw std::runtime_error(replyStr);
-            break;
+            std::string error_str = "Redis command returned an error: "+std::string(replyStr);
+            RedisModule_FreeCallReply(reply);
+            RedisModule_FreeThreadSafeContext(ctx);
+            throw std::runtime_error(error_str);
+        }
+        case REDISMODULE_REPLY_UNKNOWN: {
+            std::string error_str = "Redis command returned an error: "+std::string(strerror(errno));
```

Comment:
what does `strerror(errno)` do?

Reply:
It returns the error string associated with the given `errno` (I noticed that when `REDISMODULE_REPLY_UNKNOWN` is returned. redis sets the global `errno` to the appropriate error number).

Reply:
do you have a test for it?

Reply:
yes, it is one of the tests that I added